### PR TITLE
Use pytest-runner only for testing

### LINKED ISSUE #91: Make pytest-runner a dependency only for tests
**Describe exactly what you would like to see in an upcoming release**

Our Conda recipe for cylc-uiserver has pytest-runner as a requirement, even though we are not running tests there. We need to copy the approach from metomi-isodatetime here.

**Additional context**

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by kinow**
These changes close #91

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (why? change in build script).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

**Comment by kinow** [comment]
An easy one to be reviewed @cylc/core :grimacing:

File: setup.py [hunk 1]
```diff
@@ -44,9 +45,11 @@ def find_version(*file_paths):
     'cylc-flow==8.0a1'
 ]
 
-setup_requires = [
-    'pytest-runner==4.4.*'
-]
+# Only include pytest-runner in setup_requires if we're invoking tests
+if {'pytest', 'test', 'ptr'}.intersection(sys.argv):
```
Comment (hjoliver) [inline setup.py]: Ooh, I've actually never used `intersection()` - I haven't lived!

Comment (kinow) [reply setup.py]: Same for me. When I saw that in the PR for isodatetime from the Conda Forge reviewer I had to google it and read the documentation :)

Comment (oliver-sanders) [reply setup.py]:
```python
>>> {1, 2}.intersection({2, 3})
{2}
>>> {1, 2} & {2, 3}
{2}
```

**Pull request merged**